File: server/src/main/java/com/fluffy/exam/domain/dto/SubmittedExamSummaryDto.java
package com.fluffy.exam.domain.dto;

import com.querydsl.core.annotations.QueryProjection;
import java.time.LocalDateTime;

public record SubmittedExamSummaryDto(
        Long examId,
        String title,
        String description,
        AuthorDto author,
        Long submissionCount,
        LocalDateTime lastSubmissionDate
) {

    @QueryProjection
    public SubmittedExamSummaryDto {
    }
}

File: server/src/main/kotlin/com/fluffy/exam/domain/dto/SubmittedExamSummaryDto.kt
package com.fluffy.exam.domain.dto

import java.time.LocalDateTime

data class SubmittedExamSummaryDto(
    val examId: Long,
    val title: String,
    val description: String,
    val author: AuthorDto,
    val submissionCount: Long,
    val lastSubmissionDate: LocalDateTime,
)
